include/oneapi/dpl/pstl/hetero/dpcpp/unseq_backend_sycl.h

```diff
@@ -531,7 +540,8 @@ struct __scan
             __k = 2;
             do
             {
-                auto __shifted_local_id = __local_id - __local_id % __k - 1;
+                // use signed type to avoid overflowing
+                ::std::int32_t __shifted_local_id = __local_id - __local_id % __k - 1;
```

Comment:
Why not `int64__t`?

Reply:
It seems excessive, because  `__local_id` is limited by work group size. `__shifted_local_id` is less than `__local_id`